### PR TITLE
[MODULAR] Fixes syndicate ghost roles not spawning with codespeak language, & an assortment of language related fixes/improvements

### LINKED ISSUE #17849: DS-2/Interdyne Spawner Issues
## Round ID: 
7114

## Reproduction:

DS-2 and Interdyne ghost role players do not spawn in having the Codespeak language known. Furthermore, when they spawn in, their loadouts are given to them repeatedly, overwriting character loadouts and overflowing their storage.

### LINKED ISSUE #19575: Ashwalkers spawning without language
Reporting client version: 514.1589

## Round ID:
7755

## Testmerges:
- [[TESTMERGE CANDIDATE] [SEMI-MODULAR] AI and hologram QoL stuff + misc bugfixes](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19248)
- [[TM ONLY] Disables Sideplane, Makes the Game Playable](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19308)
- [Antag Revamp*: The Nuclear-ish Option](https://github.com/Skyrat-SS13/Skyrat-tg/pull/18110)
- [You can no longer see the icon of languages you don't know](https://github.com/Skyrat-SS13/Skyrat-tg/pull/17248)
- [Makes CMGs in the armory spawn as gunsets that include both non-lethal and lethal magazines](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19489)
- [[MIRROR] Refactors crew records (#72725) + Medical/Security records now use the max/min age. [MDB IGNORE]](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19078)
- [fixes ghost cafe sleepers](https://github.com/Skyrat-SS13/Skyrat-tg/pull/19536)

## Reproduction:
When spawning in from observation to an Ashwalker, they sometimes appear without a language. This can occur several rounds in a row or may only occur once every few rounds. The only known way to fix this issue in-game is to have an admin add back the language manually per round. 

A hypothesis presented is that it has to do with using a character loadout since in the character creation, Ashwalkers cannot select any languages.

### PR DESCRIPTION
**Pull request by vinylspiders**
## About The Pull Request

Fixes #17849
Fixes #19575

This issue occurs whenever you opt to use your preferences for spawning.

The problem is that the `special()` proc is always called before the prefs get applied, causing the language granted in `special()` to be overwritten by character pref languages.

This is basically unavoidable without some major code restructuring/skyrat edits so I just added a new proc, `post_transfer_prefs()`, which can be overridden whenever you want to set things after prefs have been applied for a particular ghost role.

Also fixed related issues for the NRI raiders and any other relevant spawners I could find, and gave them actual codenames instead of just randomized last names.

Also fixes a tangential (but not quite related) issue for ashwalkers, which was caused by the sanitize function removing it after applying prefs due to it being considered a 'secret' language.

## How This Contributes To The Skyrat Roleplay Experience

No more being forced to use a random human spawn for DS2 roles if you want to be able to use codespeak.

Edit:

Since this has been open so long, I decided to push the additional language fix I came up with with this PR.
The short of it, is it attempts to fix a bug where species-innate languages will still get applied even if you don't have them in your prefs. 

This bug has apparently kept cropping up multiple times due to the way language is coded on tg's side. It is firmly coupled with mind code, such that it is in my opinion not worth it to try to hook skyrat edits into the procs. It would just probably lead to more breakage.

My solution is to instead make use of the `COMSIG_GLOB_CREWMEMBER_JOINED` signal and apply the language prefs at the last stages of character creation, to ensure that it doesn't get overridden by any weirdness. This in my opinion is a much cleaner and reliable solution and should not break unless TG does something bad.

## Proof of Testing

<details>
<summary>Human with languages in their prefs will now properly receive codespeak on top of their pref languages</summary>
  
image

</details>

<details>
<summary>NRI raiders, same thing</summary>
  
image

</details>

<details>
<summary>Updated language perk tooltips for all species to reflect being able to choose other languages via prefs</summary>

image

</details>

<details>
<summary>Ashwalker language tooltip</summary>
  
image

</details>

<details>
<summary>Ashtongue now shows up for Ashwalkers, though it cannot be removed</summary>

image

</details>

<details>
<summary>Secret languages still remain hidden for other species and will be sanitized if incorrectly set up in prefs</summary>
  

https://user-images.githubusercontent.com/13398309/224391850-3ad29a17-f01f-4eca-b245-914f25eb52ec.mp4

</details>

## Changelog

:cl:
fix: fixes syndicate ghost roles not spawning with codespeak language when they use their character preferences
fix: fixes NRI raiders spawning without the correct language and with the wrong names on their ID cards
fix: fixes ashwalkers sometimes spawning without a language
fix: fixes common (Sol) being granted despite not having them in your prefs, once again allowing you to not be able to speak common by preference.
fix: some improvements to the language UI -- the native language perk tooltip text has been updated to reflect that it is optional to pick the species' native language. Also, ashtongue now shows up in the UI correctly for ashwalkers.
code: new codename generation for the NRI roles
/:cl:

### CODE REVIEW

**Comment by vinylspiders** [comment]
Added new name generation at @Stalkeros2 request, in prep for the move towards giving NRI their own special antag datum ( #19380 )

**Comment by Stalkeros2** [comment]
I agree. 👍

File: modular_skyrat/master_files/code/modules/mob_spawn/mob_spawn.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/mob_spawn/mob_spawn.dm:46]: Just how we typically do empty procs, leaving it fully empty makes it look like an accident.
```suggestion
/obj/effect/mob_spawn/ghost_role/proc/post_transfer_prefs(mob/living/new_spawn)
	return
```

File: modular_skyrat/modules/customization/strings/names/phonetic_alphabet_numbers.txt
Comment (GoldenAlpharex) [inline modular_skyrat/modules/customization/strings/names/phonetic_alphabet_numbers.txt:10]: Why is it Niner and not Nine?

File: modular_skyrat/modules/encounters/code/nri_raiders.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/encounters/code/nri_raiders.dm]: Those two outfits could probably be made into like `/datum/outfit/pirate/nri/officer` and `/marine` and the likes, to avoid all the duplicate stuff that's repeated, like the `post_equip` overload.

File: modular_skyrat/modules/encounters/code/nri_raiders.dm
Comment (GoldenAlpharex) [inline modular_skyrat/modules/encounters/code/nri_raiders.dm]: Why have it in `special()` and in `post_transfer_prefs()` at the same time?

Also don't add `return ..()` at the end, it'll cause issues.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
After looking at this for a little longer, this just feels like a bandaid over `special()` not always being called when it should always be called. _That_ is the issue you'll actually want to fix.

File: modular_skyrat/modules/encounters/code/nri_raiders.dm
Comment (vinylspiders) [reply modular_skyrat/modules/encounters/code/nri_raiders.dm]: Because post_transfer prefs doesn't get called if the user elects to not use their prefs the way the spawner code is written. Only special() does.

The other thing is a mistake though.

File: modular_skyrat/modules/customization/strings/names/phonetic_alphabet_numbers.txt
Comment (vinylspiders) [reply modular_skyrat/modules/customization/strings/names/phonetic_alphabet_numbers.txt:10]: Phonetic alphabet numbers

File: modular_skyrat/modules/encounters/code/nri_raiders.dm
Comment (vinylspiders) [reply modular_skyrat/modules/encounters/code/nri_raiders.dm]: Done

**Comment by vinylspiders** [comment]
> After looking at this for a little longer, this just feels like a bandaid over `special()` not always being called when it should always be called. _That_ is the issue you'll actually want to fix.

The issue isn't that special isn't being called--it always gets called. The issue is that some of the things that it sets (such as names, languages) are being overwritten by 

https://github.com/Skyrat-SS13/Skyrat-tg/blob/f70c0e0f10927f044b366ebaaa55e9f399e84510/modular_skyrat/master_files/code/modules/mob_spawn/mob_spawn.dm#L16

So anywhere special() is called these things will just end up being overwritten to be the user prefs because our special gets called after the TG special(). 

This is a consequence of the way that our mob spawning is overridden and woven into the TG one, and this is the best I could come up with short of rewriting the whole thing (which I am not willing to do in this instance).

**Review by tf-4 — APPROVED** [review]

File: modular_skyrat/modules/customization/strings/names/phonetic_alphabet_numbers.txt
Comment (Stalkeros2) [reply modular_skyrat/modules/customization/strings/names/phonetic_alphabet_numbers.txt:10]: After thinking this through, this *really* did feel better when I was thinking of it rather than when it's executed; I'll probably come up with a better system of RNG numbers at a later date.

**Comment by tf-4** [comment]
@GoldenAlpharex

**Comment by tf-4** [comment]
@GoldenAlpharex

File: code/modules/language/language_holder.dm
Comment (vinylspiders) [inline code/modules/language/language_holder.dm:54]: These skyrat edits in language_holder were nonfunctional and are no longer needed.

File: code/modules/mob_spawn/ghost_roles/mining_roles.dm
Comment (GoldenAlpharex) [inline code/modules/mob_spawn/ghost_roles/mining_roles.dm]: Add the original line after a `ORIGINAL: originalCodeHere` at the end of the comment please :)

File: code/modules/mob_spawn/ghost_roles/unused_roles.dm
Comment (GoldenAlpharex) [inline code/modules/mob_spawn/ghost_roles/unused_roles.dm]: Ditto.

File: modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/client/preferences/middleware/languages.dm]: Since you're there, yeet this. We're in modular files.
```suggestion
/datum/preference_middleware/languages/apply_to_human(mob/living/carbon/human/target, datum/preferences/preferences)
```

File: modular_skyrat/master_files/code/modules/language/language.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/language/language.dm]: Document this variable please.

File: modular_skyrat/master_files/code/modules/language/language_holder.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/modules/language/language_holder.dm:50]: Does it really have to be formatted all weird like this?

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]

File: modular_skyrat/master_files/code/modules/language/language_holder.dm
Comment (vinylspiders) [reply modular_skyrat/master_files/code/modules/language/language_holder.dm:50]: This was copied and pasted from the TG language_holder.dm

File: modular_skyrat/master_files/code/modules/language/language.dm
Comment (vinylspiders) [reply modular_skyrat/master_files/code/modules/language/language.dm]: Done

File: code/modules/mob_spawn/ghost_roles/mining_roles.dm
Comment (vinylspiders) [reply code/modules/mob_spawn/ghost_roles/mining_roles.dm]: Done

File: code/modules/mob_spawn/ghost_roles/unused_roles.dm
Comment (vinylspiders) [reply code/modules/mob_spawn/ghost_roles/unused_roles.dm]: Done

File: modular_skyrat/master_files/code/modules/language/language_holder.dm
Comment (vinylspiders) [reply modular_skyrat/master_files/code/modules/language/language_holder.dm:50]: https://github.com/Skyrat-SS13/Skyrat-tg/blob/076e3fd31fe0e8d092937430a5f1ede4422039fa/code/modules/language/language_holder.dm#L258-L264

Just did it for consistency's purposes since it is a 'master file'. I can get rid of it if you want.

**Review by GoldenAlpharex — APPROVED** [review]
Looks good to me.

**Review by Gandalf2k15 — APPROVED** [review]

**Comment by vinylspiders** [comment]
Sorry @Gandalf2k15 I was -just- in the middle of updating a quick addendum to this. You can see the commit right after your approval, and there are new screenshots in the description for the changes. Mostly just UI related cosmetic ones.

**Review by Gandalf2k15 — APPROVED** [review]

**Pull request merged**
